packages/terra-notification-dialog/src/NotificationDialog.jsx

```diff
@@ -2,96 +2,81 @@ import React from 'react';
 import PropTypes from 'prop-types';
 import AbstractModal from 'terra-abstract-modal';
 import FocusTrap from 'focus-trap-react';
-import * as KeyCode from 'keycode-js';
 import Button, { ButtonVariants } from 'terra-button';
-import classNames from 'classnames/bind';
+import classNames from 'classnames';
+import classNamesBind from 'classnames/bind';
 import { FormattedMessage } from 'react-intl';
+import ThemeContext from 'terra-theme-context';
+
+import NotificationIcon from './_NotificationIcon';
+import ContentLayoutAsList from './_ContentLayoutAsList';
+
 import styles from './NotificationDialog.module.scss';
 
-const cx = classNames.bind(styles);
+const cx = classNamesBind.bind(styles);
 
-const variants = {
-  ALERT: 'alert',
-  ERROR: 'error',
-  WARNING: 'warning',
-  INFO: 'info',
-  SUCCESS: 'success',
-  CUSTOM: 'custom',
-};
+const variants = [
+  'alert',
+  'error',
+  'warning',
+  'info',
+  'custom',
+];
 
 const propTypes = {
   /**
-   * Header of the notification-dialog.
+   * The variant of notification to be rendered. This renders the dialog with the corresponding header and icon to the
+   * variant concept.
+   * Use one of `alert`, `error`, `warning`, `info`, or `custom`.
    */
-  header: PropTypes.string,
+  variant: PropTypes.oneOf(variants).isRequired,
   /**
-   * Title of the notification-dialog.
+   * The title to describe the high-level overview of why the notification-dialog is being displayed to the user.
    */
   title: PropTypes.string,
```

Comment:
Per https://github.com/cerner/terra-core/issues/2770 - we need to not continue to use prop of `title` - not that it would be common for a consumer to attempt to spread the title attribute onto this component to add a tooltip. 
Regardless, we need to avoid potential conflict with any html attribute that might otherwise may be needed for A11Y or similar, and should be reserved for only when Terra has the same usage as the browser, rather than repurposing. But generally will be ideal to standardize a new common prop name for heading labels/title text/etc. 
 
```suggestion
  notificationTitle: PropTypes.string,
```
or 
```suggestion
  titleText: PropTypes.string,
```

Reply:
Confirmed with Matt, Tyler & Ryan - we all believe we can make this change in a way that we do not cause a cascading MVB in terra-application. I updated this prop to be `dialogTitle` 

https://github.com/cerner/terra-framework/pull/1105/commits/6a2745410a4cad58062870d21d421f0e6f55f7aa